### PR TITLE
Fix payment history API `earliest_first` parameter and compare transactions on ledger order

### PR DESCRIPTION
**Pull request by lumberj**
Fix bug in earliest_first query parameter for Payment History API

### CODE REVIEW

File: test/unit/utils-test.js [hunk 1]
```diff
@@ -234,6 +235,19 @@ suite('unit - convertAmount()', function() {
 });
 
 suite('unit - utils.compareTransactions()', function() {
+
+  function toString(val) {
+    if (typeof val === 'number') {
+      return val.toString();
+    } else if (typeof val === 'object') {
+      return _.mapValues(val, toString);
```
Comment (clark800) [inline test/unit/utils-test.js]: `toString` returns non-string return type. Could make `toStringValues` a recursive function

File: api/transactions.js [hunk 2]
```diff
@@ -607,7 +570,7 @@ function getAccountTransactions(api, options, callback) {
 
   function sortTransactions(transactions, async_callback) {
     transactions.sort(function(first, second) {
-      return compareTransactions(first, second, options.earliestFirst);
+      return utils.compareTransactions(first, second, !options.earliestFirst);
```
Comment (clark800) [inline api/transactions.js]: `var compare = options.earliestFirst ? utils.compareTransactions : _.rearg(utils.comareTransactions, 1, 0);`
`transactions.sort(compare);`
could be used to eliminate some redundancy in `compareTransactions`

Comment (lumberj) [inline api/transactions.js]: Looks good, thanks for the tip

Comment (lumberj) [inline api/transactions.js]: That's not in lodash v2

Comment (lumberj) [inline api/transactions.js]: Can we keep this as a TODO after we update lodash

File: test/unit/utils-test.js (hunk 1 above)
Comment (lumberj) [inline test/unit/utils-test.js]: done

File: api/transactions.js (hunk 2 above)
Comment (clark800) [inline api/transactions.js]: Could do that, or just do it manually:
`function swapArguments(f) { return function(a, b) { return f(b, a); } }`
or just multiply the result of the compare function by `1` or `-1`

Comment (lumberj) [inline api/transactions.js]: I'll wait for #351  to merge, and update

**Comment by lumberj** [comment]
@clark800 updated

**Comment by clark800** [comment]
LGTM

**Pull request merged**